src/ast/ast_build_filter_tree.c

```diff
@@ -69,13 +72,51 @@ FT_FilterNode *_CreateFilterSubtree(RecordMap *record_map, AST_Operator op,
  * WHERE a.val = b.val */
 FT_FilterNode *_convertBinaryOperator(RecordMap *record_map, const cypher_astnode_t *op_node) {
 	const cypher_operator_t *operator = cypher_ast_binary_operator_get_operator(op_node);
-	// Arguments are of type CYPHER_AST_EXPRESSION
-	const cypher_astnode_t *lhs = cypher_ast_binary_operator_get_argument1(op_node);
-	const cypher_astnode_t *rhs = cypher_ast_binary_operator_get_argument2(op_node);
+	AST_Operator op = AST_ConvertOperatorNode(operator);
+	const cypher_astnode_t *lhs;
+	const cypher_astnode_t *rhs;
+
+	switch(op) {
+	case OP_OR:
+	case OP_AND:
+	case OP_NOT:
+	case OP_EQUAL:
+	case OP_NEQUAL:
+	case OP_LT:
+	case OP_LE:
+	case OP_GT:
+	case OP_GE:
+		// Arguments are of type CYPHER_AST_EXPRESSION
+		lhs = cypher_ast_binary_operator_get_argument1(op_node);
+		rhs = cypher_ast_binary_operator_get_argument2(op_node);
+		return _CreateFilterSubtree(record_map, op, lhs, rhs);
+	default:
+		return FilterTree_CreateExpressionFilter(AR_EXP_FromExpression(record_map, op_node));
+	}
+}
 
+FT_FilterNode *_convertUnaryOperator(RecordMap *record_map, const cypher_astnode_t *op_node) {
+	const cypher_operator_t *operator = cypher_ast_unary_operator_get_operator(op_node);
+	// Argument is of type CYPHER_AST_EXPRESSION
+	const cypher_astnode_t *arg = cypher_ast_unary_operator_get_argument(op_node);
 	AST_Operator op = AST_ConvertOperatorNode(operator);
 
-	return _CreateFilterSubtree(record_map, op, lhs, rhs);
+	return _CreateFilterSubtree(record_map, op, arg, NULL);
+}
+
+FT_FilterNode *_convertTrueOperator() {
+	AR_ExpNode *exp = AR_EXP_NewConstOperandNode(SI_BoolVal(true));
+	return FilterTree_CreateExpressionFilter(exp);
+}
+
+FT_FilterNode *_convertFalseOperator() {
+	AR_ExpNode *exp = AR_EXP_NewConstOperandNode(SI_BoolVal(false));
+	return FilterTree_CreateExpressionFilter(exp);
+}
+
+FT_FilterNode *_convertIntegerOperator(RecordMap *record_map, const cypher_astnode_t *expr) {
```

Comment:
All these _convert routines can be static, right?

Reply:
👍